embassy-stm32/src/gpio.rs

```diff
@@ -51,8 +53,7 @@ impl<'d> Flex<'d> {
                     Pull::None => vals::CnfIn::FLOATING,
                 };
 
-                let crlh = if n < 8 { 0 } else { 1 };
-                r.cr(crlh).modify(|w| {
+                r.cr(n / 8).modify(|w| {
```

Comment:
I checked that rustc should be able to infer that `n / 8 < 2` and skip the bounds check in `r.cr(...)`.